### ISSUE TITLE
generate_TFrecord.py

### ISSUE THREAD

**Issue by jp3spinelli** [tensorflow/tensorflow#41416]
I am using my mac, running Catalina 10.15.5, with Python 3.8.3, and TF (2.2.0) which I pip3 installed from github. I have come to the conclusion the issue is using TF 2 instead of 1, but I do not want to downgrade my TF. Additionally, I named some folders differently than him i.e. his "images" is my "Camera pictures," and his "test" and "train" are my "Test" and "Train"

I've been dealing with an issue converting my csv files to TFrecord, following this tutorial https://pythonprogramming.net/creating-tfrecord-files-tensorflow-object-detection-api-tutorial/

Here is the issue I keep running into:
![image](https://user-images.githubusercontent.com/68125365/87552143-46d91700-c67f-11ea-900f-5f80329358e9.png)

My directory looks like:

Desktop

- models
- Object-Detection
        - data/
            -- Test_labels.csv
            -- Train_labels.csv
        - Camera pictures/
            -- Test/
            --- testingimages.jpg
            -- Train/
            --- testingimages.jpg
            -- ...myimages.jpg
        - training
        - xml_to_csv.py
        - generate_TFrecord.py'

Here is my generate_TFrecord.py code: 
    
    """   
    Usage:
    # From tensorflow/models/
    # Create train data:
    python3 generate_tfrecord.py --csv_input=data/Test_labels.csv --output_path=data/test.record --image_dir=Camera Pictures/Test
    # Create test data:
    python3 generate_tfrecord.py --csv_input=data/Train_labels.csv --output_path=data/train.record --image_dir=Camera Pictures/Train 
    """

    from __future__ import division
    from __future__ import print_function
    from __future__ import absolute_import

    import os
    import io
    import pandas as pd
    import tensorflow.compat.v1 as tf

    from PIL import Image
    from object_detection.utils import dataset_util
    from collections import namedtuple, OrderedDict

    flags = tf.app.flags
    flags.DEFINE_string('csv_input', '', 'Path to the CSV input')
    flags.DEFINE_string('output_path', '', 'Path to output TFRecord')
    flags.DEFINE_string('image_dir', '', 'Path to the images')
    FLAGS = flags.FLAGS

    # TO-DO replace this with label map
    def class_text_to_int(row_label):
        if row_label == 'Raspi':
            return 1
        else:
            None

    def split(df, group):
        data = namedtuple('data', ['filename', 'object'])
        gb = df.groupby(group)
        return [data(filename, gb.get_group(x)) for filename, x in zip(gb.groups.keys(), gb.groups)]

    def create_tf_example(group, path):
        with tf.gfile.GFile(os.path.join(path, '{}'.format(group.filename)), 'rb') as fid:
            encoded_jpg = fid.read()
        encoded_jpg_io = io.BytesIO(encoded_jpg)
        image = Image.open(encoded_jpg_io)
        width, height = image.size

        filename = group.filename.encode('utf8')
        image_format = b'jpg'
        xmins = []
        xmaxs = []
        ymins = []
        ymaxs = []
        classes_text = []
        classes = []

        for index, row in group.object.iterrows():
            xmins.append(row['xmin'] / width)
            xmaxs.append(row['xmax'] / width)
            ymins.append(row['ymin'] / height)
            ymaxs.append(row['ymax'] / height)
            classes_text.append(row['class'].encode('utf8'))
            classes.append(class_text_to_int(row['class']))

        tf_example = tf.train.Example(features=tf.train.Features(feature={
            'image/height': dataset_util.int64_feature(height),
            'image/width': dataset_util.int64_feature(width),
            'image/filename': dataset_util.bytes_feature(filename),
            'image/source_id': dataset_util.bytes_feature(filename),
            'image/encoded': dataset_util.bytes_feature(encoded_jpg),
            'image/format': dataset_util.bytes_feature(image_format),
            'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
            'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
            'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
            'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
            'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
            'image/object/class/label': dataset_util.int64_list_feature(classes),
        }))
        return tf_example

    def main(_):
        writer = tf.python_io.TFRecordWriter(FLAGS.output_path)
        path = os.path.join(FLAGS.image_dir)
        examples = pd.read_csv(FLAGS.csv_input)
        grouped = split(examples, 'filename')
        for group in grouped:
            tf_example = create_tf_example(group, path)
            writer.write(tf_example.SerializeToString())

        writer.close()
        output_path = os.path.join(os.getcwd(), FLAGS.output_path)
        print('Successfully created the TFRecords: {}'.format(output_path))

    if __name__ == '__main__':
        tf.app.run()

Please respond with exactly what needs to be changed in the code, as I am new to this. Thank you!

**Comment by jp3spinelli** [comment]
I asked this question on Stack Overflow but have not received a response yet, so I came here. Additionally, how do I fix the issue with my computer looking in the "Camera" folder when it should be "Camera pictures"?

**Comment by amahendrakar** [comment]
@jp3spinelli,
Could you please zip the contents of `models` and `Object-Detection` folders and share it with us so that we can look into this issue. Thanks!

**Comment by jp3spinelli** [comment]
I resolved the issue. Found a solution in Youtube comments. Thank you!
